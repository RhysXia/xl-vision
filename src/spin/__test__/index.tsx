import { mount } from 'enzyme'
import React from 'react'
import Spin from '..'
import FasCircleNotch from '../../icon/icons/fas-circle-notch'

describe('spin', () => {
  beforeAll(() => {
    jest.useRealTimers()
  })
  it('基本用法', () => {
    const wrapper = mount(<Spin />)
    expect(wrapper).toMatchSnapshot()
  })

  it('内嵌内容', () => {
    const wrapper = mount(
      <Spin>
        <div className='inner'>Further details about the context of this alert.</div>
      </Spin>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('覆盖父元素', () => {
    const wrapper = mount(
      <div className='inner'>
        Further details about the context of this alert.
        <Spin cover={true} />
      </div>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('不同大小', () => {
    const wrapper = mount(
      <div className='wrapper'>
        <Spin size='small' />
        <Spin size='default' />
        <Spin size='large' />
      </div>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('自定义描述', () => {
    const wrapper = mount(<Spin tip='Loading...' />)
    expect(wrapper).toMatchSnapshot()
  })

  it('自定义指示器', () => {
    const wrapper = mount(<Spin indicator={<FasCircleNotch spin={true} />} tip='Loading...' />)
    expect(wrapper).toMatchSnapshot()
  })

  test.todo('启动加载和关闭加载')
  // it('启动加载和关闭加载', async () => {
  //   const wrapper = mount(
  //     <Spin spinning={true}>
  //       <div className='inner'>Further details about the context of this alert.</div>
  //     </Spin>
  //   )
  //   expect(wrapper.find('.xl-spin--spinning')).not.toBeNull()
  //   wrapper.setProps({
  //     spinning: false
  //   })
  //   wrapper.update()
  //   await wait(100)
  //   console.log(wrapper.debug())
  //   expect(wrapper.find('.xl-spin--spinning')).toBeNull()
  // })
})
