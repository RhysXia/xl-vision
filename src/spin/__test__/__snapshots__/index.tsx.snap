// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spin 不同大小 1`] = `
<div
  className="wrapper"
>
  <Spin
    size="small"
  >
    <div
      className="xl-spin xl-spin--spinning"
    >
      <CssTransition
        classNames="xl-spin--fade"
        show={true}
      >
        <Transition
          afterAppear={[Function]}
          afterEnter={[Function]}
          afterLeave={[Function]}
          appear={[Function]}
          appearCancelled={[Function]}
          beforeAppear={[Function]}
          beforeEnter={[Function]}
          beforeLeave={[Function]}
          enter={[Function]}
          enterCancelled={[Function]}
          leave={[Function]}
          leaveCancelled={[Function]}
          show={true}
        >
          <div
            className="xl-spin--fade"
            style={Object {}}
          >
            <div
              className="xl-spin__wrap xl-spin__wrap--small"
            >
              <span
                className="xl-spin__indicator"
              >
                <FasSpinner
                  spin={true}
                >
                  <Memo(BaseIcon)
                    spin={true}
                  >
                    <i
                      className="xl-icon xl-icon--spin"
                      style={Object {}}
                    >
                      <svg
                        fill="currentColor"
                        style={
                          Object {
                            "height": "1em",
                            "width": "1em",
                          }
                        }
                        viewBox="0 0 512 512"
                      >
                        <path
                          d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
                        />
                      </svg>
                    </i>
                  </Memo(BaseIcon)>
                </FasSpinner>
              </span>
              <span
                className="xl-spin__tip"
              >
                loading
              </span>
            </div>
          </div>
        </Transition>
      </CssTransition>
    </div>
  </Spin>
  <Spin
    size="default"
  >
    <div
      className="xl-spin xl-spin--spinning"
    >
      <CssTransition
        classNames="xl-spin--fade"
        show={true}
      >
        <Transition
          afterAppear={[Function]}
          afterEnter={[Function]}
          afterLeave={[Function]}
          appear={[Function]}
          appearCancelled={[Function]}
          beforeAppear={[Function]}
          beforeEnter={[Function]}
          beforeLeave={[Function]}
          enter={[Function]}
          enterCancelled={[Function]}
          leave={[Function]}
          leaveCancelled={[Function]}
          show={true}
        >
          <div
            className="xl-spin--fade"
            style={Object {}}
          >
            <div
              className="xl-spin__wrap xl-spin__wrap--default"
            >
              <span
                className="xl-spin__indicator"
              >
                <FasSpinner
                  spin={true}
                >
                  <Memo(BaseIcon)
                    spin={true}
                  >
                    <i
                      className="xl-icon xl-icon--spin"
                      style={Object {}}
                    >
                      <svg
                        fill="currentColor"
                        style={
                          Object {
                            "height": "1em",
                            "width": "1em",
                          }
                        }
                        viewBox="0 0 512 512"
                      >
                        <path
                          d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
                        />
                      </svg>
                    </i>
                  </Memo(BaseIcon)>
                </FasSpinner>
              </span>
              <span
                className="xl-spin__tip"
              >
                loading
              </span>
            </div>
          </div>
        </Transition>
      </CssTransition>
    </div>
  </Spin>
  <Spin
    size="large"
  >
    <div
      className="xl-spin xl-spin--spinning"
    >
      <CssTransition
        classNames="xl-spin--fade"
        show={true}
      >
        <Transition
          afterAppear={[Function]}
          afterEnter={[Function]}
          afterLeave={[Function]}
          appear={[Function]}
          appearCancelled={[Function]}
          beforeAppear={[Function]}
          beforeEnter={[Function]}
          beforeLeave={[Function]}
          enter={[Function]}
          enterCancelled={[Function]}
          leave={[Function]}
          leaveCancelled={[Function]}
          show={true}
        >
          <div
            className="xl-spin--fade"
            style={Object {}}
          >
            <div
              className="xl-spin__wrap xl-spin__wrap--large"
            >
              <span
                className="xl-spin__indicator"
              >
                <FasSpinner
                  spin={true}
                >
                  <Memo(BaseIcon)
                    spin={true}
                  >
                    <i
                      className="xl-icon xl-icon--spin"
                      style={Object {}}
                    >
                      <svg
                        fill="currentColor"
                        style={
                          Object {
                            "height": "1em",
                            "width": "1em",
                          }
                        }
                        viewBox="0 0 512 512"
                      >
                        <path
                          d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
                        />
                      </svg>
                    </i>
                  </Memo(BaseIcon)>
                </FasSpinner>
              </span>
              <span
                className="xl-spin__tip"
              >
                loading
              </span>
            </div>
          </div>
        </Transition>
      </CssTransition>
    </div>
  </Spin>
</div>
`;

exports[`spin 内嵌内容 1`] = `
<Spin>
  <div
    className="xl-spin xl-spin--spinning xl-spin--nested"
  >
    <CssTransition
      classNames="xl-spin--fade"
      show={true}
    >
      <Transition
        afterAppear={[Function]}
        afterEnter={[Function]}
        afterLeave={[Function]}
        appear={[Function]}
        appearCancelled={[Function]}
        beforeAppear={[Function]}
        beforeEnter={[Function]}
        beforeLeave={[Function]}
        enter={[Function]}
        enterCancelled={[Function]}
        leave={[Function]}
        leaveCancelled={[Function]}
        show={true}
      >
        <div
          className="xl-spin--fade"
          style={Object {}}
        >
          <div
            className="xl-spin__wrap xl-spin__wrap--default"
          >
            <span
              className="xl-spin__indicator"
            >
              <FasSpinner
                spin={true}
              >
                <Memo(BaseIcon)
                  spin={true}
                >
                  <i
                    className="xl-icon xl-icon--spin"
                    style={Object {}}
                  >
                    <svg
                      fill="currentColor"
                      style={
                        Object {
                          "height": "1em",
                          "width": "1em",
                        }
                      }
                      viewBox="0 0 512 512"
                    >
                      <path
                        d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
                      />
                    </svg>
                  </i>
                </Memo(BaseIcon)>
              </FasSpinner>
            </span>
            <span
              className="xl-spin__tip"
            >
              loading
            </span>
          </div>
        </div>
      </Transition>
    </CssTransition>
    <div
      className="xl-spin__children"
    >
      <div
        className="inner"
      >
        Further details about the context of this alert.
      </div>
    </div>
  </div>
</Spin>
`;

exports[`spin 基本用法 1`] = `
<Spin>
  <div
    className="xl-spin xl-spin--spinning"
  >
    <CssTransition
      classNames="xl-spin--fade"
      show={true}
    >
      <Transition
        afterAppear={[Function]}
        afterEnter={[Function]}
        afterLeave={[Function]}
        appear={[Function]}
        appearCancelled={[Function]}
        beforeAppear={[Function]}
        beforeEnter={[Function]}
        beforeLeave={[Function]}
        enter={[Function]}
        enterCancelled={[Function]}
        leave={[Function]}
        leaveCancelled={[Function]}
        show={true}
      >
        <div
          className="xl-spin--fade"
          style={Object {}}
        >
          <div
            className="xl-spin__wrap xl-spin__wrap--default"
          >
            <span
              className="xl-spin__indicator"
            >
              <FasSpinner
                spin={true}
              >
                <Memo(BaseIcon)
                  spin={true}
                >
                  <i
                    className="xl-icon xl-icon--spin"
                    style={Object {}}
                  >
                    <svg
                      fill="currentColor"
                      style={
                        Object {
                          "height": "1em",
                          "width": "1em",
                        }
                      }
                      viewBox="0 0 512 512"
                    >
                      <path
                        d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
                      />
                    </svg>
                  </i>
                </Memo(BaseIcon)>
              </FasSpinner>
            </span>
            <span
              className="xl-spin__tip"
            >
              loading
            </span>
          </div>
        </div>
      </Transition>
    </CssTransition>
  </div>
</Spin>
`;

exports[`spin 自定义指示器 1`] = `
<Spin
  indicator={
    <FasCircleNotch
      spin={true}
    />
  }
  tip="Loading..."
>
  <div
    className="xl-spin xl-spin--spinning"
  >
    <CssTransition
      classNames="xl-spin--fade"
      show={true}
    >
      <Transition
        afterAppear={[Function]}
        afterEnter={[Function]}
        afterLeave={[Function]}
        appear={[Function]}
        appearCancelled={[Function]}
        beforeAppear={[Function]}
        beforeEnter={[Function]}
        beforeLeave={[Function]}
        enter={[Function]}
        enterCancelled={[Function]}
        leave={[Function]}
        leaveCancelled={[Function]}
        show={true}
      >
        <div
          className="xl-spin--fade"
          style={Object {}}
        >
          <div
            className="xl-spin__wrap xl-spin__wrap--default"
          >
            <span
              className="xl-spin__indicator"
            >
              <FasCircleNotch
                spin={true}
              >
                <Memo(BaseIcon)
                  spin={true}
                >
                  <i
                    className="xl-icon xl-icon--spin"
                    style={Object {}}
                  >
                    <svg
                      fill="currentColor"
                      style={
                        Object {
                          "height": "1em",
                          "width": "1em",
                        }
                      }
                      viewBox="0 0 512 512"
                    >
                      <path
                        d="M288 39.056v16.659c0 10.804 7.281 20.159 17.686 23.066C383.204 100.434 440 171.518 440 256c0 101.689-82.295 184-184 184-101.689 0-184-82.295-184-184 0-84.47 56.786-155.564 134.312-177.219C216.719 75.874 224 66.517 224 55.712V39.064c0-15.709-14.834-27.153-30.046-23.234C86.603 43.482 7.394 141.206 8.003 257.332c.72 137.052 111.477 246.956 248.531 246.667C393.255 503.711 504 392.788 504 256c0-115.633-79.14-212.779-186.211-240.236C302.678 11.889 288 23.456 288 39.056z"
                      />
                    </svg>
                  </i>
                </Memo(BaseIcon)>
              </FasCircleNotch>
            </span>
            <span
              className="xl-spin__tip"
            >
              Loading...
            </span>
          </div>
        </div>
      </Transition>
    </CssTransition>
  </div>
</Spin>
`;

exports[`spin 自定义描述 1`] = `
<Spin
  tip="Loading..."
>
  <div
    className="xl-spin xl-spin--spinning"
  >
    <CssTransition
      classNames="xl-spin--fade"
      show={true}
    >
      <Transition
        afterAppear={[Function]}
        afterEnter={[Function]}
        afterLeave={[Function]}
        appear={[Function]}
        appearCancelled={[Function]}
        beforeAppear={[Function]}
        beforeEnter={[Function]}
        beforeLeave={[Function]}
        enter={[Function]}
        enterCancelled={[Function]}
        leave={[Function]}
        leaveCancelled={[Function]}
        show={true}
      >
        <div
          className="xl-spin--fade"
          style={Object {}}
        >
          <div
            className="xl-spin__wrap xl-spin__wrap--default"
          >
            <span
              className="xl-spin__indicator"
            >
              <FasSpinner
                spin={true}
              >
                <Memo(BaseIcon)
                  spin={true}
                >
                  <i
                    className="xl-icon xl-icon--spin"
                    style={Object {}}
                  >
                    <svg
                      fill="currentColor"
                      style={
                        Object {
                          "height": "1em",
                          "width": "1em",
                        }
                      }
                      viewBox="0 0 512 512"
                    >
                      <path
                        d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
                      />
                    </svg>
                  </i>
                </Memo(BaseIcon)>
              </FasSpinner>
            </span>
            <span
              className="xl-spin__tip"
            >
              Loading...
            </span>
          </div>
        </div>
      </Transition>
    </CssTransition>
  </div>
</Spin>
`;

exports[`spin 覆盖父元素 1`] = `
<div
  className="inner"
>
  Further details about the context of this alert.
  <Spin
    cover={true}
  >
    <div
      className="xl-spin xl-spin--spinning xl-spin--cover"
    >
      <CssTransition
        classNames="xl-spin--fade"
        show={true}
      >
        <Transition
          afterAppear={[Function]}
          afterEnter={[Function]}
          afterLeave={[Function]}
          appear={[Function]}
          appearCancelled={[Function]}
          beforeAppear={[Function]}
          beforeEnter={[Function]}
          beforeLeave={[Function]}
          enter={[Function]}
          enterCancelled={[Function]}
          leave={[Function]}
          leaveCancelled={[Function]}
          show={true}
        >
          <div
            className="xl-spin--fade"
            style={Object {}}
          >
            <div
              className="xl-spin__wrap xl-spin__wrap--default"
            >
              <span
                className="xl-spin__indicator"
              >
                <FasSpinner
                  spin={true}
                >
                  <Memo(BaseIcon)
                    spin={true}
                  >
                    <i
                      className="xl-icon xl-icon--spin"
                      style={Object {}}
                    >
                      <svg
                        fill="currentColor"
                        style={
                          Object {
                            "height": "1em",
                            "width": "1em",
                          }
                        }
                        viewBox="0 0 512 512"
                      >
                        <path
                          d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
                        />
                      </svg>
                    </i>
                  </Memo(BaseIcon)>
                </FasSpinner>
              </span>
              <span
                className="xl-spin__tip"
              >
                loading
              </span>
            </div>
          </div>
        </Transition>
      </CssTransition>
    </div>
  </Spin>
</div>
`;
